docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -0,0 +1,121 @@
+# New Language Feature Check Integration Process [WIP]
+
+## Check Update Procedure
+
+### Consider Similar Tokens
+
+Identify existing tokens that are similar to the new language feature tokens.
+This comparison helps to determine which existing checks may be relevant
+and require updates to support the new token.
+
+**Examples**:
+
+- When the `RECORD_DEF` token was introduced, it was reasonable to look at all checks
+  that had `CLASS_DEF` in their acceptable tokens.
+
+- When `TEXT_BLOCK_CONTENT` token was introduced, it was reasonable to look at all checks
+  that had `STRING_LITERAL` in their acceptable tokens.
+
+### Java Enhancement Proposals (JEPS)
+
+Review Java Enhancement Proposals (JEPS) related to the new feature.
+JEPS provide detailed information about the goals
+and motivations behind new language features.
+This information can help us understand the feature better and figure out
+which checks are most likely to be impacted.
+
+**Examples**:
+
+- Unnamed variables (`_`), JEP guides us to recognize the new role of the
+  underscore for unnamed variables and avoid flagging the non-use of such variables.
+  Consequently, we should update `UnusedLocalVariableCheck` to ensure it does not
+  incorrectly violate unnamed variables.
+
+### Always Impacted Checks
+
+Certain checks are most likely to be impacted by new language features.
+It is mandatory to analyze impact on the following checks:
+
+- [IllegalToken](https://checkstyle.org/checks/coding/illegaltoken.html)
+- [IllegalTokenText](https://checkstyle.org/checks/coding/illegaltokentext.html)
+- [Indentation](https://checkstyle.org/checks/misc/indentation.html#Indentation)
+- [Whitespace](https://checkstyle.org/checks/whitespace/index.html)
+
+### Review Other Static Analysis Tools
+
+Examining how other static analysis tools handle new language features
+can provide valuable insights into best practices and potential pitfalls.
+
+**Examples**:
+
+- IntelliJ IDEA introduced an inspection rule for Java 17 or higher to detect
+  redundant `strictfp` modifiers. This resulted in analysis of `RedundantModifierCheck`
+  to verify if we needed to update it.
+
+### Real Usages Examples in Large Projects
+
+Review real usage examples of the new language feature in large projects.
+This helps to identify potential issues, and ensures that the checks
+are aligned with practical use cases.
+List of representative projects can be found
+[here](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/github-action-projects1.properties)
+
+## New Check Procedure
```

Comment:
Add 1-2 sentences here about how creating new checks for new language features helps us to establish best practices within the community.